File: hibernate/src/main/java/io/github/robertomike/hefesto/actions/wheres/WhereField.java
package io.github.robertomike.hefesto.actions.wheres;

import io.github.robertomike.hefesto.enums.Operator;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class WhereField extends Where {
    private String parentField;

    public WhereField(String field, String parentField) {
        super(field);
        this.parentField = parentField;
    }

    public WhereField(String field, Operator operator, String parentField) {
        super(field);
        this.operator = operator;
        this.parentField = parentField;
    }
}

File: hibernate/src/main/kotlin/io/github/robertomike/hefesto/actions/wheres/WhereField.kt
package io.github.robertomike.hefesto.actions.wheres

import io.github.robertomike.hefesto.enums.Operator

class WhereField : Where {
    var parentField: String? = null

    constructor(field: String, parentField: String) : super(field) {
        this.parentField = parentField
    }

    constructor(field: String, operator: Operator, parentField: String) : super(field) {
        this.operator = operator
        this.parentField = parentField
    }
}
